tsl/src/chunk_api.c

```diff
@@ -371,3 +372,118 @@ chunk_create(PG_FUNCTION_ARGS)
 
 	PG_RETURN_DATUM(HeapTupleGetDatum(tuple));
 }
+
+/*
+ * Detach a chunk from a hypertable.
+ */
+Datum
+chunk_detach(PG_FUNCTION_ARGS)
+{
+	Oid chunk_relid = PG_ARGISNULL(0) ? InvalidOid : PG_GETARG_OID(0);
+	Cache *hcache;
+	Hypertable *ht;
+	Chunk *chunk;
+
+	TS_PREVENT_FUNC_IF_READ_ONLY();
+
+	chunk = ts_chunk_get_by_relid(chunk_relid, true);
+	Assert(chunk != NULL);
+
+	ht = ts_hypertable_cache_get_cache_and_entry(chunk->hypertable_relid, CACHE_FLAG_NONE, &hcache);
+	Assert(ht != NULL);
+
+	if (!object_ownercheck(RelationRelationId, chunk_relid, GetUserId()))
+		aclcheck_error(ACLCHECK_NOT_OWNER,
+					   get_relkind_objtype(chunk->relkind),
+					   get_rel_name(chunk_relid));
+
+	if (ts_chunk_is_compressed(chunk))
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot detach compressed chunk \"%s\"", get_rel_name(chunk_relid)),
+				 errhint("decompress the chunk first")));
```

Comment:
Unlike main message, the hint should start capitalized and end with a period.
```suggestion
				 errhint("Decompress the chunk first.")));
```